src/commonMain/kotlin/fr/acinq/lightning/io/Peer.kt

```diff
@@ -708,6 +707,32 @@ class Peer(
         peerConnection?.send(message)
     }
 
+    sealed class SelectChannelResult {
+        /** We have a channel that is available for payments and splicing. */
+        data class Available(val channel: Normal) : SelectChannelResult()
+        /** We have a channel, or will have one soon, but it is not ready yet. */
+        data object NotReady : SelectChannelResult()
+        /** We don't have any channel, or our channel is closing, so we need a new one. */
+        data object None : SelectChannelResult()
+    }
+
+    private fun selectChannelForSplicing(): SelectChannelResult = when (val channel = channels.values.firstOrNull { it is Normal }) {
+        is Normal -> when {
+            channel.spliceStatus == SpliceStatus.None -> SelectChannelResult.Available(channel)
+            else -> SelectChannelResult.NotReady
+        }
+        else -> when {
+            // We are currently funding a new channel, which should be available soon.
+            channels.values.any { it is WaitForOpenChannel || it is WaitForAcceptChannel || it is WaitForFundingCreated || it is WaitForFundingSigned } -> SelectChannelResult.NotReady
+            // We have funded a channel and are waiting for it to be ready.
+            channels.values.any { it is WaitForFundingConfirmed || it is WaitForChannelReady } -> SelectChannelResult.NotReady
+            // All channels are closing, we need a new one.
+            channels.values.all { it is ShuttingDown || it is Negotiating || it is Closing || it is Closed || it is Aborted } -> SelectChannelResult.None
+            // Note that we don't check for Offline or Syncing, because we wait for channel reestablishment already in [waitForPeerReady].
```

Comment:
I don't know if we should still check for `Offline` and `Syncing`, I'm afraid we'd risk hitting the bug mentioned in `waitForPeerReady` and would thus incorrectly return `SelectChannelResult.NotReady` and wait forever...or maybe we should use the same logic that `waitForPeerReady` contains, which would make `selectChannelForSplicing` more exhaustive?

Reply:
Done in https://github.com/ACINQ/lightning-kmp/pull/659/commits/d39bc01a8ce020799dd685e58c0d3bfe3228d8d1.

I'll try an alternative way of fixing this issue, by having `lightning-kmp` send its own `channel_reestablish` if it doesn't receive one after 10 seconds, this way the remote node can answer with an `error` and we will clean-up the channel.

Reply:
The syncing channels issue is fixed in #664